sys/net/network_layer/fib/fib.c

```diff
@@ -421,6 +421,41 @@ int fib_get_next_hop(kernel_pid_t *iface_id,
     return 0;
 }
 
+int fib_get_destination_set(uint32_t bitmap, uint32_t type,
+                            fib_destination_set_entry_t *dst_set, size_t* dst_set_size)
+{
+    mutex_lock(&mtx_access);
+    int ret = -EHOSTUNREACH;
+    size_t found_entries = 0;
+
+    for (size_t i = 0; i < FIB_MAX_FIB_TABLE_ENTRIES; ++i) {
+        if( (fib_table[i].global_flags & bitmap) == (type & bitmap) ) {
+
+            if( dst_set != NULL && found_entries < (*dst_set_size)-1 ) {
```

Comment:
Added parens and wrote a line that the matching count is being determined.

The idea to allow a `dst_set == NULL` and continue, is that if you're interested at first in the current number of matching entries, to check if your out buffer can hold them, you probably don't want to have any copy operations.